### PR TITLE
Fixes #11792: Generic method file_ensure_key_value_present_in_ini_section.cf has leaky classes, resulting in invalid results

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/11792

### CODE REVIEW

File: tests/acceptance/30_generic_methods/file_ensure_key_value_present_in_ini_section.cf [hunk 1]
```diff
@@ -52,13 +52,14 @@ ${base_text_down}";
...
+# Third test: ensure that if the line already exists, it will be modified (must have an end section to work with cfengine <3.10)
     "section_3"         string => "section_test1";
     "name_3"            string => "name";
     "value_3"           string => "value";
     "new_value_3"       string => "newvalue";
-    "reference_3"       string => "${base_text_up}
-${name_3}=${new_value_3}";
+    "reference_3"       string => "[${section_3}]
+${name_3}=${new_value_3}
+[endfile]";
```
Comment (amousset) [inline tests/acceptance/30_generic_methods/file_ensure_key_value_present_in_ini_section.cf]: We should not require this, we have an exeption in stdlib to handle the behavior change.

Comment (ncharles) [reply tests/acceptance/30_generic_methods/file_ensure_key_value_present_in_ini_section.cf]: it does not work on cfengine 3.7 without this

Comment (ncharles) [reply tests/acceptance/30_generic_methods/file_ensure_key_value_present_in_ini_section.cf]: https://tracker.mender.io/browse/CFE-2263?jql=text%20~%20%22select_region%22

Comment (ncharles) [reply tests/acceptance/30_generic_methods/file_ensure_key_value_present_in_ini_section.cf]: it actually fails from 3.6.3 to 3.8 according to ticket

Comment (amousset) [reply tests/acceptance/30_generic_methods/file_ensure_key_value_present_in_ini_section.cf]: How could this test past before in this case?

Comment (ncharles) [reply tests/acceptance/30_generic_methods/file_ensure_key_value_present_in_ini_section.cf]: we only have agent in 3.6.2 and 3.10 if i'm correct

Comment (amousset) [reply tests/acceptance/30_generic_methods/file_ensure_key_value_present_in_ini_section.cf]: We have 3.6.5 in 3.1, and we had 3.7 in 3.2/4.0

Comment (ncharles) [reply tests/acceptance/30_generic_methods/file_ensure_key_value_present_in_ini_section.cf]: you are perfectly right, I was testing with a vanilla 3.7

**Comment by ncharles** [comment]
Commit modified

File: tests/acceptance/30_generic_methods/file_ensure_key_value_present_in_ini_section.cf [hunk 2]
```diff
@@ -57,7 +57,7 @@ ${base_text_up}";
     "name_3"            string => "name";
     "value_3"           string => "value";
     "new_value_3"       string => "newvalue";
-    "reference_3"       string => "${base_text_up}
+    "reference_3"       string => "[${section_3}]
```
Comment (amousset) [inline tests/acceptance/30_generic_methods/file_ensure_key_value_present_in_ini_section.cf]: why this change?

Comment (ncharles) [reply tests/acceptance/30_generic_methods/file_ensure_key_value_present_in_ini_section.cf]: for readability

Comment (ncharles) [reply tests/acceptance/30_generic_methods/file_ensure_key_value_present_in_ini_section.cf]: i scratched my head thrice over the test of why it was section1 in this test, so had to look up to base_text_up, which is not informative

**Comment by ncharles** [comment]
OK, merging this PR

**Comment by ncharles** [comment]
OK, merging this PR
